internal/services/compute/snapshot_data_source.go

```diff
@@ -120,44 +119,59 @@ func dataSourceSnapshotRead(d *pluginsdk.ResourceData, meta interface{}) error {
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id := parse.NewSnapshotID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
+	id := snapshots.NewSnapshotID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	resp, err := client.Get(ctx, id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			return fmt.Errorf("%s was not found", id)
 		}
 		return fmt.Errorf("loading %s: %+v", id, err)
 	}
 
 	d.SetId(id.ID())
 
-	if props := resp.SnapshotProperties; props != nil {
-		d.Set("os_type", string(props.OsType))
-		d.Set("time_created", props.TimeCreated.String())
+	if model := resp.Model; model != nil {
+		if props := model.Properties; props != nil {
+			if props.OsType != nil {
+				d.Set("os_type", string(*props.OsType))
+			}
 
-		if props.DiskSizeGB != nil {
-			d.Set("disk_size_gb", int(*props.DiskSizeGB))
-		}
+			if props.TimeCreated != nil {
+				d.Set("time_created", *props.TimeCreated)
+			}
 
-		if err := d.Set("encryption_settings", flattenSnapshotDiskEncryptionSettings(props.EncryptionSettingsCollection)); err != nil {
-			return fmt.Errorf("setting `encryption_settings`: %+v", err)
-		}
+			if props.DiskSizeGB != nil {
+				d.Set("disk_size_gb", int(*props.DiskSizeGB))
+			}
 
-		trustedLaunchEnabled := false
-		if securityProfile := props.SecurityProfile; securityProfile != nil {
-			if securityProfile.SecurityType == compute.DiskSecurityTypesTrustedLaunch {
-				trustedLaunchEnabled = true
+			if err := d.Set("encryption_settings", flattenSnapshotDiskEncryptionSettings(props.EncryptionSettingsCollection)); err != nil {
+				return fmt.Errorf("setting `encryption_settings`: %+v", err)
+			}
+
+			trustedLaunchEnabled := false
+			if securityProfile := props.SecurityProfile; securityProfile != nil && securityProfile.SecurityType != nil {
+				if *securityProfile.SecurityType == snapshots.DiskSecurityTypesTrustedLaunch {
+					trustedLaunchEnabled = true
+				}
```

Comment:
fwiw this can become:

```suggestion
				trustedLaunchEnabled = *securityProfile.SecurityType == snapshots.DiskSecurityTypesTrustedLaunch
```